drivers/ma_reports/app/models/ma_reports/csg_engage/credential.rb

```diff
@@ -0,0 +1,60 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'uri'
+require 'net/http'
+require 'net/http/post/multipart'
+module MaReports::CsgEngage
+  class Credential < ::GrdaWarehouse::RemoteCredential
+    # Docs: TODO
+    alias_attribute :apikey, :password
+
+    def post(body)
+      url = URI("#{endpoint}/Import")
+      https = Net::HTTP.new(url.host, url.port)
+      https.use_ssl = true
+      file = Multipart::Post::UploadIO.new(StringIO.new(body.to_json), 'application/json', 'test.json')
+
+      request = Net::HTTP::Post::Multipart.new(url, file: file)
```

Comment:
Totally fine to use what you have, but if you find the API nicer (probably not significantly so) we do have the [Curb gem](https://github.com/taf2/curb).

Reply:
Gotcha, thanks. I'm going to leave this as is, since it is currently working , but I'll keep that in mind for the future